x-pack/plugins/logs_shared/public/components/log_ai_assistant/log_ai_assistant.tsx

```diff
@@ -24,13 +23,15 @@ export interface LogAIAssistantDocument {
 
 export interface LogAIAssistantProps {
   doc: LogAIAssistantDocument | undefined;
+  ContextualInsight: ObservabilityAIAssistantPluginStart['ContextualInsight'];
 }
 
 export interface LogAIAssistantDeps extends LogAIAssistantProps {
   observabilityAIAssistant: ObservabilityAIAssistantPluginStart['service'];
+  ContextualInsight: ObservabilityAIAssistantPluginStart['ContextualInsight'];
 }
 
-export const LogAIAssistant = withProviders(({ doc }: LogAIAssistantProps) => {
+export const LogAIAssistant = withProviders(({ doc, ContextualInsight }: LogAIAssistantProps) => {
```

Comment:
@marcogavaz As per our Slack conversation, I believe this construction needs to be simplified so type errors are correctly reported during dev time.

Reply:
@CoenWarmer I believe you meant to tag me on this one, I'll take a look thanks!

Reply:
I created this [tech debt issue](https://github.com/elastic/kibana/issues/173493) to avoid generating conflicts on this work, let me know if the workaround you mentioned offline works for this change or if we need to tackle this sooner.

Reply:
Apologies @tonyghiani, indeed! I've since rested a bit and came up with a solution that I believe works and simplifies things a bit. Let me know what you think!